services/grpc/internal/transformations/graph_objects.go

```diff
@@ -0,0 +1,36 @@
+package transformations
+
+type Node struct {
+	NodeId string `db:"nodeId"`
+	Doc string `db:"doc"`
+
+	EdgesOut string[] `db:"edgesOut"` // IDs of edges whose destination is this node
+	EdgesIn string[] `db:"edgesIn"`   // ID of edges whose source is this node
+
+	// Metadata
+	Date string `db:"date"`
+	Author string `db:"author"`
```

Comment:
It may be good to add a version number to the Node, so that new metadata attributes can be added over time.

Reply:
Another approach is to allow metadata to be extensible and allow empty values, because old Nodes will then default to empty. Personally I don't like this as much because history is being subtly modified.

Reply:
I need to do some reading on how to gracefully version data.

Reply:
TODO: I will make a version metadata struct and add a version

Reply:
Also add option fields like counter for debugging purposes.

Reply:
Also add an array of scope ids.